tests/kernel/userbufer_validate/src/main.c

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2017 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ztest.h>
+extern void test_buffer_rw_read(void);
+extern void test_buffer_writeable_write(void);
+extern void test_buffer_readable_read(void);
+extern void test_buffer_readable_write(void);
+extern void test_buffer_supervisor_rw(void);
+extern void test_buffer_supervisor_w(void);
+extern void test_buffer_user_rw_user(void);
+extern void test_buffer_user_rw_supervisor(void);
+/*test case nain entry*/
+void test_main(void *p1, void *p2, void *p3)
```

Comment:
why is this in a separate file???

Reply:
This will make the code more legible and easy. Also some of the other codes follow same.